static-build/qemu-virtiofs/Dockerfile

```diff
@@ -19,37 +19,56 @@ RUN apt-get --no-install-recommends install -y \
 	    flex \
 	    gawk \
 	    libaudit-dev \
+	    libblkid-dev \
 	    libcap-dev \
 	    libcap-ng-dev \
 	    libdw-dev \
 	    libelf-dev \
+	    libffi-dev \
 	    libglib2.0-0 \
 	    libglib2.0-dev \
 	    libglib2.0-dev git \
 	    libltdl-dev \
+	    libmount-dev \
 	    libpixman-1-dev \
 	    libpmem-dev \
 	    libseccomp-dev \
+	    libseccomp2 \
+	    libselinux1-dev \
 	    libtool \
-	    patch \
+	    make \
 	    pkg-config \
 	    pkg-config \
 	    python \
 	    python-dev \
 	    rsync \
 	    seccomp \
-	    libseccomp2 \
 	    zlib1g-dev
 
 RUN cd  .. && git clone "${QEMU_VIRTIOFS_REPO}" qemu-virtiofs
 RUN git checkout "${QEMU_VIRTIOFS_TAG}"
-ADD qemu/patches/virtiofsd/0001-add-time-to-seccomp.patch /root/0001-add-time-to-seccomp.patch
-ADD qemu/patches/virtiofsd/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch /root/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch
-RUN patch -p1 < /root/0001-add-time-to-seccomp.patch
-RUN patch -p1 < /root/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch
+
 ADD scripts/configure-hypervisor.sh /root/configure-hypervisor.sh
-RUN PREFIX="${PREFIX}" /root/configure-hypervisor.sh -s kata-qemu | sed -e 's|--enable-rbd||g' -e 's|--disable-seccomp||g' | xargs ./configure \
+ADD qemu/patches/ /root/kata_qemu_patches
+# Apply virtiofs specific patches
+RUN  \
+	for patch in $(find /root/kata_qemu_patches/virtiofsd/${QEMU_VIRTIOFS_TAG}/ -name '*.patch'); do\
+		echo "apply $patch";\
+		git apply "$patch"; \
+	done
+
+# Apply general kata patches for the qemu version
+RUN  \
+	cat VERSION; \
+	stable_branch=$(cat VERSION | awk 'BEGIN{FS=OFS="."}{print $1 "." $2 ".x"}');\
+	for patch in $(find /root/kata_qemu_patches/${stable_branch}/ -name '*.patch'); do\
```

Comment:
As soon as you start doing a for loop in a Dockerfile, you know you've gone too far man!

I'll see if I can provide a better suggestion in a bit, but this shouldn't live in the Dockerfile.

Reply:
:) yeah does not look nice, I have tried to change to way to do it, please take a look or suggest any other approach